testapp/src/main/java/com/rakuten/tech/mobile/testapp/analytics/rat_wrapper/RATEvent.kt

```diff
@@ -28,28 +32,39 @@ class RATEvent {
         return action ?: ActionType.OPEN
     }
 
-    fun getLabel(): String {
-        return label ?: ""
+    fun getPgn():String{
+        return "$siteSection→$pageName"
     }
 
-    fun getMiniAppInfo(): MiniAppInfo? {
-        return miniAppInfo
+    fun getTargetElement(): String {
+        return targetElement ?: ""
     }
 
-    constructor(event: EventType, action: ActionType) {
+    constructor(event: EventType, pageName: String, siteSection: String){
         this.event = event
-        this.action = action
+        this.pageName = pageName
+        this.siteSection = siteSection
     }
 
-    constructor(event: EventType, action: ActionType, label: String) {
+    constructor(event: EventType,action: ActionType, pageName: String, siteSection: String, componentName: String, elementType: String){
         this.event = event
-        this.action = action
-        this.label = label
+        this.action= action
+        this.pageName = pageName
+        this.siteSection = siteSection
+        this.componentName = componentName
+        this.elementType = elementType
+        //{screen_name}:{component_name}-{element_type}.{action}
+        this.targetElement = "$pageName: $componentName-$elementType.${action.value}"
     }
 
-    constructor(event: EventType, action: ActionType, miniAppInfo: MiniAppInfo) {
+    constructor(event: EventType, pageName: String, siteSection: String, componentName: String, elementType: String){
         this.event = event
-        this.action = action
-        this.miniAppInfo = miniAppInfo
+        this.pageName = pageName
+        this.siteSection = siteSection
+        this.componentName = componentName
+        this.elementType = elementType
+        //{screen_name}:{component_name}-{element_type}
+        this.targetElement = "$pageName: $componentName-$elementType"
```

Comment:
`pageName` is sent as a separate field, so you don't need to include it in the `targetElement`